### ISSUE TITLE
AIVs have limited amount of raid catapults

### ISSUE THREAD

**Issue by Admiral-Gumpert** [Sh0wdown/UnofficialCrusaderPatch#371]
I just observed that every AIV after the latest patch won't spam the catapults or fire ballistas with the exact same amount he can do from the start of the match  : here's an example :- 
![image](https://user-images.githubusercontent.com/50601775/57661419-6e53d000-75f3-11e9-9464-f5ca313505c4.png)  This is The Wazir  As you can see he recruited 8 FireBallistas at one,,, but after a while he will start to reduce the numbers until it reaches ZERO , Yes after a while he will really stop making them at all and that also applies to all the Characters , any suggestions to solve this?

**Comment by Sh0wdown** [comment]
That's not caused by the patch but actually a vanilla bug. I think in theory it happens when the AI builds something (like another siege tent) on top of the engines. This makes the siege engine vanish, while the AI still thinks it is there, stopping it from rebuilding it.

**Comment by Panbutt** [comment]
Yeah about 70% of the time the AI will keep placing tents on top of harassing siege engines before said engine can move out of the way. I'm trying to figure out if there's some sort of delay field that determines the speed of which the tents are placed.

**Comment by Admiral-Gumpert** [comment]
Idk if this is really the issue , In the screen shot above it shows he managed to recruit all of the 8 ballistas but after a while this number will reduce to 6 then 2 then 1 and then disappear , meaning he only made a total of 30 ballistas or so

**Comment by Admiral-Gumpert** [comment]
We just need a tool or a program to extract files from older versions of stronghold crusader

**Comment by Halbarad1** [comment]
> That's not caused by the patch but actually a vanilla bug. I think in theory it happens when the AI builds something (like another siege tent) on top of the engines. This makes the siege engine vanish, while the AI still thinks it is there, stopping it from rebuilding it.

So could it be possible to stop that? At the moment, AI also doesn't build its buildings on players troops so maybe there could be an exception for their siege engines?

**Comment by Heroesflorian** [comment]
To be honest, I kinda doubt that "building stuff over siege engines" is the (full/only) reason. I've already had matches where Saladin (vanilla, i.e. maximum of 5 concurrent harassment catapults) built tents over his harassment catapults dozens of times and still built more... so I think there must be something else at work (additionally).

**Comment by GutentagderHerr** [comment]
The best thing would be: The KI cant "overbuild" units or trees like a Real Player. That would solve many Problems. I cant forget the Image of Marschall deleting his Knights because of a Dairy Farm :D

**Comment by DarkThief-Darek** [comment]
> The best thing would be: The KI cant "overbuild" units or trees like a Real Player. That would solve many Problems. I cant forget the Image of Marschall deleting his Knights because of a Dairy Farm :D

That's a really bad idea in my eyes. Because if a tower/gatehouse is manned and destroyed (without the ruins) the troops also still alive with half HP. If the AI can't build over those troops he can't rebuild his lost towers/gates until all his defensive troops on that stop are dead.

**Comment by Truetobi91** [comment]
Is this problem still being worked on? I can approve that the AI can overbuild more than 8 single harrass siege engines and still continious to build them. But outside of that I'd still love to have the problem solved so that the AI doesn't spend thousands on gold for overbuilding their fkn siege engines.

**Comment by Admiral-Gumpert** [comment]
> Is this problem still being worked on? I can approve that the AI can overbuild more than 8 single harrass siege engines and still continious to build them. But outside of that I'd still love to have the problem solved so that the AI doesn't spend thousands on gold for overbuilding their fkn siege engines.

For some reason AIs in my game wont continue to build those siege engines untill all engineers are killed!! Hope for a solution as this is one of the biggest weakpoints for AIs
